---
 man/dmsetup.8.in |    7 ++++
 tools/dmsetup.c  |   88 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 95 insertions(+)

--- LVM2.2.02.132/man/dmsetup.8.in.orig	2015-10-25 08:37:19.863432358 +0100
+++ LVM2.2.02.132/man/dmsetup.8.in	2015-10-25 08:51:30.263414633 +0100
@@ -23,6 +23,9 @@
 .IR options ]
 .RI [ device_name ]
 .HP
+.B dmsetup export
+.RI [ device_name ]
+.HP
 .B dmsetup help
 .RB [ \-c | \-C | \-\-columns ]
 .HP
@@ -354,6 +357,11 @@
 \fBblkdevname\fP (block device name),
 \fBdevname\fP (map name for device-mapper devices, equal to blkdevname otherwise).
 .HP
+.B export
+.RI [ device_name ]
+.br
+Outputs information in key/value format to be imported by other programs.
+.HP
 .B help
 .RB [ \-c | \-C | \-\-columns ]
 .br
--- LVM2.2.02.130/tools/dmsetup.c.orig	2015-09-09 18:44:39.471523160 +0200
+++ LVM2.2.02.130/tools/dmsetup.c	2015-09-09 18:48:33.918179988 +0200
@@ -2189,6 +2189,93 @@
 	return r;
 }
 
+static int _export(CMD_ARGS)
+{
+	int r = 0;
+	struct dm_task *dmt = NULL;
+	void *next = NULL;
+	uint64_t start, length;
+	char *target_type = NULL;
+	char *params;
+	const char *name = NULL;
+	const char *uuid = NULL;
+	struct dm_info info;
+
+	if (names)
+		name = names->name;
+	else if (argc == 2)
+		name = argv[1];
+
+	if (!(dmt = dm_task_create(DM_DEVICE_STATUS)))
+		goto out;
+
+	if (!_set_task_device(dmt, name, 0))
+		goto out;
+
+	if (!dm_task_run(dmt))
+		goto out;
+
+	if (!dm_task_get_info(dmt, &info) || !info.exists)
+		goto out;
+
+	if (!name)
+		name = dm_task_get_name(dmt);
+
+	uuid = dm_task_get_uuid(dmt);
+	printf("DM_NAME=%s\n", name);
+
+	if ((uuid = dm_task_get_uuid(dmt)) && *uuid)
+		printf("DM_UUID=%s\n", uuid);
+
+	if (!info.exists) {
+		printf("DM_STATE=NOTPRESENT\n");
+		goto out;
+	}
+
+	printf("DM_STATE=%s\n",
+	       info.suspended ? "SUSPENDED" :
+	       (info.read_only ? "READONLY" : "ACTIVE"));
+
+	if (!info.live_table && !info.inactive_table)
+		printf("DM_TABLE_STATE=NONE\n");
+	else
+		printf("DM_TABLE_STATE=%s%s%s\n",
+		       info.live_table ? "LIVE" : "",
+		       info.live_table && info.inactive_table ? "/" : "",
+		       info.inactive_table ? "INACTIVE" : "");
+
+	if (info.open_count != -1)
+		printf("DM_OPENCOUNT=%d\n", info.open_count);
+
+	printf("DM_LAST_EVENT_NR=%" PRIu32 "\n", info.event_nr);
+
+	printf("DM_MAJOR=%d\n", info.major);
+	printf("DM_MINOR=%d\n", info.minor);
+
+	if (info.target_count != -1)
+		printf("DM_TARGET_COUNT=%d\n", info.target_count);
+
+	/* export all table types */
+	next = dm_get_next_target(dmt, next, &start, &length,
+				  &target_type, &params);
+	if (target_type) {
+		printf("DM_TARGET_TYPES=%s", target_type);
+		while (next) {
+			next = dm_get_next_target(dmt, next, &start, &length,
+						  &target_type, &params);
+			if (target_type)
+				printf(",%s", target_type);
+		}
+		printf("\n");
+	}
+
+	r = 1;
+      out:
+	if (dmt)
+		dm_task_destroy(dmt);
+	return r;
+}
+
 /* Show target names and their version numbers */
 static int _targets(CMD_ARGS)
 {
@@ -5139,6 +5226,7 @@
 	{"stats", "<command> [<options>] [<devices>]", 1, -1, 1, 1, _stats},
 	{"status", "[<device>] [--noflush] [--target <target_type>]", 0, -1, 1, 0, _status},
 	{"table", "[<device>] [--target <target_type>] [--showkeys]", 0, -1, 1, 0, _status},
+	{"export", "[<device>]", 0, 1, 1, 0, _export},
 	{"wait", "<device> [<event_nr>] [--noflush]", 0, 2, 0, 0, _wait},
 	{"mknodes", "[<device>]", 0, -1, 1, 0, _mknodes},
 	{"mangle", "[<device>]", 0, -1, 1, 0, _mangle},
