diff -urN device-mapper.1.00.17.org/make.tmpl.in device-mapper.1.00.17/make.tmpl.in
--- device-mapper.1.00.17.org/make.tmpl.in	2004-04-23 19:25:37.387843152 +0200
+++ device-mapper.1.00.17/make.tmpl.in	2004-04-23 19:25:48.306183312 +0200
@@ -61,8 +61,6 @@
 
 ifeq ("@DEBUG@", "yes")
   CFLAGS += -g -fno-omit-frame-pointer -DDEBUG
-else
-  CFLAGS += -O2
 endif
 
 ifeq ("@INTL@", "yes")
--- device-mapper.1.00.17/configure.in.orig	2004-04-19 15:10:05.000000000 +0200
+++ device-mapper.1.00.17/configure.in	2004-05-06 19:11:13.292086408 +0200
@@ -48,12 +48,11 @@
 
 case "$host_os" in
 	linux*)
-		CFLAGS=
 		CLDFLAGS="-Wl,--version-script,.export.sym"
 		LD_DEPS=".export.sym"
 		SOFLAG="-shared" ;;
 	darwin*)
-		CFLAGS="-no-cpp-precomp"
+		CFLAGS="$CFLAGS -no-cpp-precomp"
 		CLDFLAGS=
 		LD_DEPS=
 		SOFLAG="-dynamiclib" ;;
