diff -urN LVM2.2.00.08.org/lib/format_text/format-text.h LVM2.2.00.08/lib/format_text/format-text.h
--- LVM2.2.00.08.org/lib/format_text/format-text.h	2004-01-25 01:02:10.990584087 +0100
+++ LVM2.2.00.08/lib/format_text/format-text.h	2004-01-25 01:09:21.088218167 +0100
@@ -7,6 +7,8 @@
 #ifndef _LVM_FORMAT_TEXT_H
 #define _LVM_FORMAT_TEXT_H
 
+#include <asm/types.h>
+
 #include "lvm-types.h"
 #include "metadata.h"
 #include "pool.h"
diff -urN LVM2.2.00.08.org/lib/label/label.h LVM2.2.00.08/lib/label/label.h
--- LVM2.2.00.08.org/lib/label/label.h	2004-01-25 01:02:11.288522184 +0100
+++ LVM2.2.00.08/lib/label/label.h	2004-01-25 01:09:35.574208118 +0100
@@ -7,6 +7,8 @@
 #ifndef _LVM_LABEL_H
 #define _LVM_LABEL_H
 
+#include <asm/types.h>
+
 #include "lvmcache.h"
 #include "uuid.h"
 #include "device.h"
