diff -dur -x '*~' LVM2.2.02.95.orig/configure.in LVM2.2.02.95/configure.in
--- LVM2.2.02.95.orig/configure.in	2012-07-01 15:33:45.000000000 +0200
+++ LVM2.2.02.95/configure.in	2012-07-02 18:41:07.560481138 +0200
@@ -1267,7 +1267,7 @@
 lvm_exec_prefix=$exec_prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$ac_default_prefix
-LVM_PATH="$lvm_exec_prefix/sbin/lvm"
+LVM_PATH="$sbindir/lvm"
 AC_DEFINE_UNQUOTED(LVM_PATH, ["$LVM_PATH"], [Path to lvm binary.])
 
 if  test "$CLVMD" != none; then
