diff -urN LVM2.2.00.09.org/lib/commands/toolcontext.c LVM2.2.00.09/lib/commands/toolcontext.c
--- LVM2.2.00.09.org/lib/commands/toolcontext.c	2004-04-01 02:11:54.589799192 +0200
+++ LVM2.2.00.09/lib/commands/toolcontext.c	2004-04-01 02:14:58.296871472 +0200
@@ -561,8 +561,10 @@
 	mallopt(M_MMAP_MAX, 0);
 #endif
 
+#ifndef INITRD_WRAPPER
 	if (!setlocale(LC_ALL, ""))
 		log_error("setlocale failed");
+#endif
 
 #ifdef INTL_PACKAGE
 	bindtextdomain(INTL_PACKAGE, LOCALEDIR);
diff -urN LVM2.2.00.09.org/tools/commands.h LVM2.2.00.09/tools/commands.h
--- LVM2.2.00.09.org/tools/commands.h	2004-04-01 02:11:55.034731552 +0200
+++ LVM2.2.00.09/tools/commands.h	2004-04-01 02:18:19.373303208 +0200
@@ -28,9 +28,11 @@
     extents_ARG, size_ARG, nofsck_ARG, test_ARG)
 *********/
 
+#ifndef INITRD_WRAPPER
 xx(dumpconfig,
    "Dump active configuration",
    "dumpconfig <filename>\n")
+#endif
 
 xx(help,
    "Display help for commands",
@@ -46,6 +48,7 @@
    "Logical Volume(s)\n")
 ***********/
 
+#ifndef INITRD_WRAPPER
 xx(lvchange,
    "Change the attributes of logical volume(s)",
    "lvchange\n"
@@ -528,6 +531,7 @@
    "\tVolumeGroupName",
 
    file_ARG, list_ARG, metadatatype_ARG, name_ARG, test_ARG)
+#endif
 
 xx(vgchange,
    "Change volume group attributes",
@@ -552,6 +556,7 @@
    logicalvolume_ARG, partial_ARG, resizeable_ARG, resizable_ARG, deltag_ARG,
    addtag_ARG, test_ARG, uuid_ARG)
 
+#ifndef INITRD_WRAPPER
 xx(vgck,
    "Check the consistency of volume group(s)",
    "vgck "
@@ -758,6 +763,7 @@
    aligned_ARG, ignorelockingfailure_ARG, noheadings_ARG, nolocking_ARG,
    nosuffix_ARG, options_ARG, partial_ARG, separator_ARG, sort_ARG,
    unbuffered_ARG, units_ARG)
+#endif
 
 xx(vgscan,
    "Search for all volume groups",
@@ -772,6 +778,7 @@
 
    ignorelockingfailure_ARG, mknodes_ARG, partial_ARG)
 
+#ifndef INITRD_WRAPPER
 xx(vgsplit,
    "Move physical volumes into a new volume group",
    "vgsplit " "\n"
@@ -787,6 +794,7 @@
    "\tPhysicalVolumePath [PhysicalVolumePath...]\n",
 
    autobackup_ARG, list_ARG, metadatatype_ARG, test_ARG)
+#endif
 
 xx(version,
    "Display software and driver version information",
diff -urN LVM2.2.00.09.org/tools/lvmcmdline.c LVM2.2.00.09/tools/lvmcmdline.c
--- LVM2.2.00.09.org/tools/lvmcmdline.c	2004-04-01 02:11:55.039730792 +0200
+++ LVM2.2.00.09/tools/lvmcmdline.c	2004-04-01 02:16:52.354532072 +0200
@@ -1299,7 +1299,11 @@
 	base = basename(namebase);
 	while (*base == '/')
 		base++;
-	if (strcmp(base, "lvm"))
+	if (strcmp(base, "lvm")
+#ifdef INITRD_WRAPPER
+		&& strcmp(base, "initrd-lvm")
+#endif
+	   )
 		alias = 1;
 	free(namebase);
 
