--- LVM2.2.00.08/make.tmpl.in.orig	2003-11-28 23:04:00.000000000 +0100
+++ LVM2.2.00.08/make.tmpl.in	2003-11-28 23:05:20.112898400 +0100
@@ -57,8 +57,6 @@
 
 ifeq ("@DEBUG@", "yes")
   CFLAGS+=-g -fno-omit-frame-pointer -DDEBUG_MEM -DDEBUG
-else
-  CFLAGS+=-O2
 endif
 
 #CFLAGS+=-DDEBUG_POOL
--- LVM2.2.00.08/configure.in.orig	2003-11-06 18:14:05.000000000 +0100
+++ LVM2.2.00.08/configure.in	2003-11-28 23:04:56.455494872 +0100
@@ -49,11 +49,6 @@
 dnl Get system type
 AC_CANONICAL_SYSTEM
 
-case "$host_os" in
-	linux*)
-		CFLAGS= ;;
-esac
-
 dnl -- prefix is /usr by default, the exec_prefix default is setup later
 AC_PREFIX_DEFAULT(/usr)
 
