--- LVM2.2.02.149/lib/replicator/replicator.c.orig	2016-04-01 21:32:16.000000000 +0200
+++ LVM2.2.02.149/lib/replicator/replicator.c	2016-04-06 18:57:18.845480954 +0200
@@ -752,8 +752,8 @@
 #ifdef REPLICATOR_INTERNAL
 int init_replicator_segtype(struct cmd_context *cmd, struct segtype_library *seglib)
 #else /* Shared */
-int init_multiple_segtype(struct cmd_context *cmd, struct segtype_library *seglib);
-int init_multiple_segtype(struct cmd_context *cmd, struct segtype_library *seglib)
+int init_multiple_segtypes(struct cmd_context *cmd, struct segtype_library *seglib);
+int init_multiple_segtypes(struct cmd_context *cmd, struct segtype_library *seglib)
 #endif
 {
 	struct segment_type *segtype;
