diff -urN LVM2.2.00.15.org/lib/commands/toolcontext.c LVM2.2.00.15/lib/commands/toolcontext.c
--- LVM2.2.00.15.org/lib/commands/toolcontext.c	2004-04-23 19:34:01.116264792 +0200
+++ LVM2.2.00.15/lib/commands/toolcontext.c	2004-04-23 19:34:08.787098648 +0200
@@ -584,8 +584,10 @@
 	mallopt(M_MMAP_MAX, 0);
 #endif
 
+#ifndef INITRD_WRAPPER
 	if (!setlocale(LC_ALL, ""))
 		log_error("setlocale failed");
+#endif
 
 #ifdef INTL_PACKAGE
 	bindtextdomain(INTL_PACKAGE, LOCALEDIR);
diff -urN LVM2.2.00.15.org/tools/lvmcmdline.c LVM2.2.00.15/tools/lvmcmdline.c
--- LVM2.2.00.15.org/tools/lvmcmdline.c	2004-04-23 19:34:01.339230896 +0200
+++ LVM2.2.00.15/tools/lvmcmdline.c	2004-04-23 19:35:45.377414688 +0200
@@ -1335,7 +1335,11 @@
 	base = basename(namebase);
 	while (*base == '/')
 		base++;
-	if (strcmp(base, "lvm") && strcmp(base, "lvm.static"))
+	if (strcmp(base, "lvm") && strcmp(base, "lvm.static")
+#ifdef INITRD_WRAPPER
+	    && strcmp(base, "initrd-lvm")
+#endif
+	)
 		alias = 1;
 	free(namebase);
 
