--- LVM2.2.02.40/configure.in.orig	2008-08-05 16:29:38.000000000 +0200
+++ LVM2.2.02.40/configure.in	2008-09-28 00:45:06.000000000 +0200
@@ -414,7 +414,7 @@
 ################################################################################
 dnl -- Check for termcap (Shamelessly copied from parted 1.4.17)
 if test x$READLINE != xno; then
-	AC_SEARCH_LIBS([tgetent], [ncurses curses termcap termlib],
+	AC_SEARCH_LIBS([tgetent], [tinfo ncurses curses termcap termlib],
 	  [tg_found=yes], [tg_found=no])
 	test x$READLINE:$tg_found = xyes:no &&
 	  AC_MSG_ERROR(
