--- LVM2.2.03.07/scripts/lvm2_activation_generator_systemd_red_hat.c~	2019-11-30 14:57:10.000000000 +0100
+++ LVM2.2.03.07/scripts/lvm2_activation_generator_systemd_red_hat.c	2019-12-17 21:15:23.770902038 +0100
@@ -145,7 +145,7 @@ static int generate_unit(struct generato
 		fprintf(f, "After=%s iscsi.service fcoe.service rbdmap.service\n"
 			"Before=remote-fs-pre.target shutdown.target\n\n"
 			"[Service]\n"
-			"ExecStartPre=/usr/bin/udevadm settle\n", _unit_names[UNIT_MAIN]);
+			"ExecStartPre=/sbin/udevadm settle\n", _unit_names[UNIT_MAIN]);
 	} else {
 		if (unit == UNIT_EARLY)
 			fputs("After=systemd-udev-settle.service\n"
