--- LVM2.2.02.85/scripts/lvm2_monitoring_init_red_hat.in.orig	2011-10-05 21:33:09.165296447 +0200
+++ LVM2.2.02.85/scripts/lvm2_monitoring_init_red_hat.in	2011-10-05 21:36:47.152774996 +0200
@@ -1,3 +1,3 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Copyright (C) 2007-2009 Red Hat, Inc. All rights reserved.
@@ -50,7 +50,8 @@ start()
 	VGSLIST=`$VGS --noheadings -o name 2> /dev/null`
 	for vg in $VGSLIST
 	do
-	    action "Starting monitoring for VG $vg:" $VGCHANGE --monitor y --poll y $vg || ret=$?
+		msg_starting "Starting monitoring for VG $vg"
+		daemon $VGCHANGE --monitor y --poll y $vg || return $?
 	done
 
 	return $ret
@@ -68,7 +71,8 @@ stop()
 	VGSLIST=`$VGS --noheadings -o name 2> /dev/null`
 	for vg in $VGSLIST
 	do
-	    action "Stopping monitoring for VG $vg:" $VGCHANGE --monitor n $vg || ret=$?
+		msg_stopping "Stopping monitoring for VG $vg"
+	    	$VGCHANGE --monitor n $vg || return $?
 	done
 	return $ret
 }
