--- LVM2.2.02.95/lib/misc/lvm-wrappers.c~	2011-04-22 14:00:00.000000000 +0200
+++ LVM2.2.02.95/lib/misc/lvm-wrappers.c	2012-06-02 17:13:13.996363539 +0200
@@ -72,7 +72,7 @@
 		return NULL;
 	}
 
-	return udev_get_dev_path(_udev);
+	return "/dev";
 }
 
 struct udev* udev_get_library_context(void)
--- LVM2.2.02.95/tools/dmsetup.c~	2012-06-02 17:11:33.999700397 +0200
+++ LVM2.2.02.95/tools/dmsetup.c	2012-06-02 17:18:14.323019614 +0200
@@ -1047,12 +1047,11 @@
 			  " defined by --udevcookie option.",
 			  _udev_cookie);
 
-	if (!(udev = udev_new()) ||
-	    !(udev_dev_dir = udev_get_dev_path(udev)) ||
-	    !*udev_dev_dir) {
-		log_error("Could not get udev dev path.");
+	if (!(udev = udev_new())) {
+		log_error("Could not get udev.");
 		return 0;
 	}
+	udev_dev_dir = "/dev";
 	udev_dev_dir_len = strlen(udev_dev_dir);
 
 	/*
